### PR TITLE
Update tooling

### PR DESCRIPTION
**Pull request by Alexandru-Dumitru**
Tools added:
- prettier
- madge
- eslint

Tools removed:
- tslint

### CODE REVIEW

**Review by zdmullen — APPROVED** [review]
Looks good!

File: .vscodeignore [hunk 1]
```diff
@@ -17,11 +17,10 @@ src/**
 .gitignore
```
Comment (VitGottwald) [inline .vscodeignore:17]: I think it is better to do it the other way around - forbidding everything and explicitly including the content we want included in the final vsix. Like we did it in Bridge for Git Explorer.

File: Use-Cases-DONT_COMMIT.md [hunk 2]
```diff
@@ -0,0 +1,3 @@
+- create profile with same name (error OK)
```
Comment (VitGottwald) [inline Use-Cases-DONT_COMMIT.md:1]: Looks like an accidentally committed file - at least according to its name.

File: tsconfig.json [hunk 3]
```diff
@@ -12,26 +12,21 @@
...
+  "compilerOptions": {
+    "module": "commonjs",
+    "target": "es6",
+    "outDir": "out",
+    "lib": ["es7"],
+    "skipLibCheck": true,
+    "sourceMap": true,
+    "strict": true /* enable all strict type-checking options */,
+    /* Additional Checks */
+    "noImplicitAny": false
```
Comment (VitGottwald) [inline tsconfig.json:25]: We should enable this ASAP.

File: tsconfig.json [hunk 4]
```diff
@@ -12,26 +12,21 @@
...
-        ".vscode-test"
-    ],
-    "include": ["src", "test", "__mocks__"]
+{
+  "compilerOptions": {
+    "module": "commonjs",
+    "target": "es6",
+    "outDir": "out",
+    "lib": ["es7"],
+    "skipLibCheck": true,
```
Comment (VitGottwald) [inline tsconfig.json:21]: And we should disable this ASAP.

**Review by VitGottwald — APPROVED** [review]

File: tsconfig.json (hunk 3 above)
Comment (Alexandru-Dumitru) [reply tsconfig.json:25]: I agree. + others. Thanks for bringing it up.

**Comment by Alexandru-Dumitru** [comment]
I have addressed some of the feedback. For the compilation options, we need a roadmap, because each action triggers errors that need to be addressed. Thanks @VitGottwald for the review ❤️

**Pull request merged**